test/unit/SmartContracts/state.unit.test.mjs

```diff
@@ -0,0 +1,1057 @@
+    ],
+    compressedSecrets: [
+      '0x0000000000000000000000000000000000000000000000000000000000000000',
+      '0x0000000000000000000000000000000000000000000000000000000000000000',
+    ],
+    proof: [
+      '0x1aceea8d85c56b2eb7eb394591d8b5da69448eff68b43f73a6535f58a39b107a',
+      '0x2238f221dcce70cbe1ca2bded0b923013148476798dc531beb93ac5498ca7bcf',
+      '0x20b3765b6a0654a77e391d4802434bea28b807441091f274448ef42d8c43a3f0',
+      '0x0ff4e034de91170aa481bcb14b0aa904f4e2961b630092e1fa79c10da20589ae',
+    ],
+  };
+
+  beforeEach(async () => {
+    [addr1, addr2, addressS, addressC] = await ethers.getSigners();
+
+    block = {
+      leafCount: 700,
+      proposer: addr1.address,
+      root: '0x2bcd2b4a55cf968f9f3dd85b3dbdd3da19e44d11a0053b221c9b5dfaf9792127',
+      blockNumberL2: 0,
+      previousBlockHash: '0x01b3dd9607d81663fc1437e08423f028070afd2006a2679b3c674f64176fd934',
+      frontierHash: '0xa2f1ec04a89542d6f1e04449398052422c7b1057df8606db047f48047bb7ab72',
+      transactionHashesRoot: '0x0487da81cb1d53536928de44fa55de0accf9a8bc9f42739a80f69584970d572f',
+    };
+
+    const Proposers = await ethers.getContractFactory('Proposers');
+    ProposersInstance = await upgrades.deployProxy(Proposers, []);
+    await ProposersInstance.deployed();
+
+    const Verifier = await ethers.getContractFactory('Verifier');
+    const verifier = await Verifier.deploy();
+    await verifier.deployed();
+
+    const Poseidon = await ethers.getContractFactory('Poseidon');
+    const poseidon = await Poseidon.deploy();
+    await poseidon.deployed();
+
+    const MerkleTree = await ethers.getContractFactory('MerkleTree_Stateless', {
+      libraries: {
+        Poseidon: poseidon.address,
+      },
+    });
+    const merkleTree = await MerkleTree.deploy();
+    await merkleTree.deployed();
+
+    const ChallengesUtil = await ethers.getContractFactory('ChallengesUtil', {
+      libraries: {
+        MerkleTree_Stateless: merkleTree.address,
+      },
+    });
+    const challengesUtil = await ChallengesUtil.deploy();
+    await challengesUtil.deployed();
+
+    const Challenges = await ethers.getContractFactory('Challenges', {
+      libraries: {
+        Verifier: verifier.address,
+        ChallengesUtil: challengesUtil.address,
+      },
+    });
+    const challenges = await upgrades.deployProxy(Challenges, [], {
+      unsafeAllow: ['external-library-linking'],
+    });
+    await challenges.deployed();
+
+    const Shield = await ethers.getContractFactory('Shield');
+    const shield = await upgrades.deployProxy(Shield, []);
+    await shield.deployed();
+
+    const Utils = await ethers.getContractFactory('Utils');
+    const utils = await Utils.deploy();
+    await utils.deployed();
+
+    const State = await ethers.getContractFactory('State', {
+      libraries: {
+        Utils: utils.address,
+      },
+    });
+    state = await upgrades.deployProxy(State, [addr1.address, addressC.address, addressS.address], {
```

Comment:
The addresses passed to the state contract should be the proposer address, challenge address and shield address

Reply:
I added different addresses from chellenge, shield, and proposer because I wanted to test individual functions directly on the state without going through other smart contract functions, avoiding potential side effects.
to test some functions, like proposeblock I mocked the data of blocks and transaions and I think I entered wrong data.
using the createBlockAndTransactions function and calling proposeBlock (not the bug fixed version, the old one) actually gives error on the last revert.
I will fix using createBlockAndTransactions for the mocked data and testing both reverts.
thanks